File: exams/villagomez/u3/examExercise/model/InsertionSort.js
export class InsertionSort {
    sort(array) {
        let n = array.length;
        for (let i = 1; i < n; i++) {
            let key = array[i];
            let j = i - 1;
            while (j >= 0 && array[j] > key) {
                array[j + 1] = array[j];
                j--;
            }
            array[j + 1] = key;
        }
        return array;
    }
}

File: hw/Rodriguez/u3/hw30/hw30Python/StrategyPattern/ec_edu_sortingapp_controller/insertion_sort.py
import sys
import os

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from sorting_strategy import SortingStrategy

class InsertionSort(SortingStrategy):
    def sort(self, data):
        n = len(data)
        for i in range(1, n):
            key = data[i]
            j = i - 1

            while j >= 0 and data[j] > key:
                data[j + 1] = data[j]
                j -= 1
            data[j + 1] = key
            
        return data
